ledger/ledger.go

```diff
@@ -469,24 +469,24 @@ func (l *Ledger) Lookup(rnd basics.Round, addr basics.Address) (basics.AccountDa
 	if err != nil {
 		return basics.AccountData{}, err
 	}
-	// todo :
-	// implement this properly so that we can take the ledgercore.AccountData, add the resources, and rebuild teh basics.AccountData.
-
-	return data, nil
+	// TODO:
+	// implement this properly so that we can take the ledgercore.AccountData, add the resources, and rebuild the basics.AccountData.
+	_ = data
+	return basics.AccountData{}, nil
 }
 
 // LookupResource loads a resource that matches the request parameters from the accounts update
-func (l *Ledger) LookupResource(rnd basics.Round, addr basics.Address, aidx basics.CreatableIndex, ctype basics.CreatableType) (ledgercore.AccountResource, error) {
+func (l *Ledger) LookupResource(rnd basics.Round, addr basics.Address, aidx basics.CreatableIndex, ctype basics.CreatableType) (ledgercore.AccountResource, basics.Round, error) {
```

Comment:
You might want to drop the validThrough here. I don't believe it would be needed for resources.
( unless you think otheewise - please let me know )

Reply:
I guess most of the LookupWithoutRewards users ignore the validThrough too..

Reply:
So, having this on the `LookupWithoutRewards` was meant for the agreement service - so that it could be caching the AccountData for a long duration without going to the ledger ( keep in mind that it was polling the data for round x-320 ). For the resources, we always need round x-1, so I don't believe this would be of value.

Reply:
ohhh OK. I removed!